kratos/python_scripts/fix_scalar_variable_process.py

```diff
@@ -52,6 +53,9 @@ def ExecuteInitializeSolutionStep(self):
             is_fixed = self.settings["constrained"].GetBool()
             if is_fixed:
                 self.variable_utils.ApplyFixity(self.variable, is_fixed, self.model_part.Nodes)
+            else:
+                if self.settings["free_dof_if_specified"].GetBool():
+                    self.variable_utils.ApplyFixity(self.variable, is_fixed, self.model_part.Nodes)
```

Comment:
```suggestion
            self.variable_utils.ApplyFixity(self.variable, is_fixed, self.model_part.Nodes)
```

wouldn't this do the same?

Reply:
yes, but it would change the previous behavior. That's what I did locally

Reply:
what is the expected behavior if `contrained` is set to `false`?

Reply:
it will not fix it